collect_app/src/main/java/org/odk/collect/android/widgets/video/VideoWidgetAnswerViewModel.kt

```diff
@@ -0,0 +1,67 @@
+package org.odk.collect.android.widgets.video
+
+import android.content.Context
+import android.graphics.Bitmap
+import android.media.MediaMetadataRetriever
+import android.net.Uri
+import androidx.compose.ui.graphics.ImageBitmap
+import androidx.compose.ui.graphics.asImageBitmap
+import androidx.core.net.toUri
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import org.odk.collect.android.utilities.MediaUtils
+import org.odk.collect.android.utilities.QuestionMediaManager
+import timber.log.Timber
+import java.io.File
+import kotlin.use
+
+class VideoWidgetAnswerViewModel(
+    private val questionMediaManager: QuestionMediaManager,
+    private val mediaUtils: MediaUtils
+) : ViewModel() {
+
+    fun init(answer: String?, context: Context): StateFlow<ImageBitmap?> {
+        val bitmapState = MutableStateFlow<ImageBitmap?>(null)
+
+        val file = questionMediaManager.getAnswerFile(answer)
+        if (file != null) {
+            viewModelScope.launch(Dispatchers.IO) {
```

Comment:
We've avoided using Coroutines directly and preferred to use them opaquely through `Scheduler`. Do you think we should discuss moving away from that approach?

Reply:
>  Do you think we should discuss moving away from that approach?

Maybe not as a part of this pr. We don't use `viewModelScope` at all so let's use scheduler here too. But yeah maybe we should file an issue to investigate and discuss it.